thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyMergeExecutor.java

```diff
@@ -184,128 +179,55 @@ private void updateMergedScoreAndPersist(MergedAnomalyResultDTO mergedResult) {
     }
   }
 
-  // TODO: Update severity according to history data
-  private void updateMergedSeverity(MergedAnomalyResultDTO anomalyMergedResult) throws Exception {
+  private void updateMergedAnomalyWeight(MergedAnomalyResultDTO anomalyMergedResult) throws Exception {
     AnomalyFunctionDTO anomalyFunctionSpec = anomalyMergedResult.getFunction();
+    BaseAnomalyFunction anomalyFunction = anomalyFunctionFactory.fromSpec(anomalyFunctionSpec);
 
-    // create time series request
-    TimeSeriesRequest timeSeriesRequest = new TimeSeriesRequest();
-    timeSeriesRequest.setCollectionName(anomalyFunctionSpec.getCollection());
-
-    List<MetricExpression> metricExpressions = Utils
-        .convertToMetricExpressions(anomalyFunctionSpec.getMetric(),
-            anomalyFunctionSpec.getMetricFunction(), anomalyFunctionSpec.getCollection());
+    List<Pair<Long, Long>> startEndTimeRanges =
+        anomalyFunction.getDataRangeIntervals(anomalyMergedResult.getStartTime(), anomalyMergedResult.getEndTime());
 
-    timeSeriesRequest.setMetricExpressions(metricExpressions);
-    TimeGranularity timeBucket = new TimeGranularity(anomalyFunctionSpec.getBucketSize(),
+    TimeGranularity timeGranularity = new TimeGranularity(anomalyFunctionSpec.getBucketSize(),
         anomalyFunctionSpec.getBucketUnit());
-    timeSeriesRequest.setAggregationTimeGranularity(timeBucket);
 
-    timeSeriesRequest.setEndDateInclusive(false);
+    MetricTimeSeries metricTimeSeries =
+        TimeSeriesUtil.getTimeSeriesResponseForPresentation(anomalyFunctionSpec, startEndTimeRanges,
```

Comment:
if we are using this for both presentation(UI) and background job, we should rename getTimeSeriesResponseForPresentation method